### PR TITLE
Decouple api-server from controller framework

### PR DESCRIPTION
**Pull request by leonwanghui**
**What this PR does / why we need it**:
See [design spec](https://github.com/opensds/design-specs/pull/17) for detailed information.
Please notice that this PR would address `Phase 1` of the design spec.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
@xing-yang @wisererik Please take a review at this PR, thanks!

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by wisererik** [comment]
please fix the CI error.

**Comment by leonwanghui** [comment]
Currently this PR has passed unit test and integration test, @xxwjj please take a look at it and update `devsds` script, thanks!

**Comment by leonwanghui** [comment]
Hi @xing-yang @wisererik , please take a review at this patch, thanks!

File: cmd/osdsapiserver/osdsapiserver.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	// Load global configuration from specified config file.
+	def := GetDefaultConfig()
+
+	flag := &CONF.Flag
+	flag.StringVar(&CONF.OsdsApiServer.ApiEndpoint, "api-endpoint", def.OsdsApiServer.ApiEndpoint, "Listen endpoint of api-server service")
+	flag.DurationVar(&CONF.OsdsApiServer.LogFlushFrequency, "log-flush-frequency", def.OsdsApiServer.LogFlushFrequency, "Maximum number of seconds between log flushes")
+	flag.BoolVar(&CONF.OsdsApiServer.Daemon, "daemon", def.OsdsApiServer.Daemon, "Run app as a daemon with -daemon=true")
+	flag.StringVar(&CONF.Database.Endpoint, "db-endpoint", def.Database.Endpoint, "Connection endpoint of database service")
+	flag.StringVar(&CONF.Database.Driver, "db-driver", def.Database.Driver, "Driver name of database service")
+	CONF.Load("/etc/opensds/opensds.conf")
```
Comment (xing-yang) [inline cmd/osdsapiserver/osdsapiserver.go]: Config file path should not be hard-coded.

File: pkg/api/volumeGroup.go [hunk 2]
```diff
@@ -77,34 +112,83 @@ func (v *VolumeGroupPortal) UpdateVolumeGroup() {
...
+			model.ErrorInternalServer, err)
 		return
 	}
+
 	v.SuccessHandle(StatusAccepted, nil)
+
+	// NOTE:The real volume group deletion process.
+	// Volume group deletion request is sent to the Dock. Dock will remove
+	// volume group record after volume group deletion operation is completed.
+	if err = v.CtrClient.Connect(CONF.OsdsLet.ApiEndpoint); err != nil {
```
Comment (xing-yang) [inline pkg/api/volumeGroup.go]: We should let controller update the database, not Dock.

File: cmd/osdsapiserver/osdsapiserver.go (hunk 1 above)
Comment (leonwanghui) [reply cmd/osdsapiserver/osdsapiserver.go]: Got it

File: pkg/api/volumeGroup.go (hunk 2 above)
Comment (leonwanghui) [reply pkg/api/volumeGroup.go]: Ok, I will update the code in next PR.

**Comment by leonwanghui** [comment]
@xing-yang Please take a look at this patch, thanks!

**Comment by leonwanghui** [comment]
I will merge this patch for now, and start the second step of [design proposal](https://github.com/opensds/design-specs/pull/17)

**Pull request merged**